lesson22/main.js

```diff
@@ -145,56 +145,85 @@ const createSortBtn = () => {
   return btn;
 };
 
-const renderSortBtn = () => {
-  const btn = createSortBtn();
-  const targetEle = document.getElementById("id");
-  targetEle.appendChild(btn);
+const setDataSort = (usersData) => {
+  Object.keys(usersData[0]).forEach((key) => {
+    if (targetSortElements.includes(key)) {
+      const element = document.getElementById(key);
+      element.dataset.sort = "sort";
+    }
+  });
+};
+
+const renderSortBtn = (usersData) => {
+  setDataSort(usersData);
+  const elements = document.querySelectorAll("[data-sort]");
+  Array.from(elements).forEach((element) => {
+    const btn = createSortBtn();
+    element.appendChild(btn);
+  });
 };
 
 const clickSortBtn = (usersData) => {
-  const sortArrow = document.querySelector(".sort-btn");
-  const sortArrowImg = document.querySelector(".sort-img");
-  sortArrow.addEventListener("click", () => {
-    changeSortStateAndArrowImg(sortArrowImg);
-    rerenderTableData(usersData);
+  const sortArrowBtns = document.querySelectorAll(".sort-btn");
+  Array.from(sortArrowBtns).forEach((sortArrowBtn) => {
+    const tHeaderId = sortArrowBtn.parentNode.id;
+    const sortArrowImg = sortArrowBtn.firstElementChild;
+    sortArrowBtn.addEventListener("click", () => {
```

Comment:
use event object.  instead of `sortArrowBtn.parentNode.id;`
event.currentTarget.parentNode.id

Reply:
In this part, I wasn't sure which way to implement them, but I will use the event object from now and then.
Thank you for teaching me.
[123dbde](https://github.com/yuka830/js-lessons/pull/28/commits/123dbdef3fe81a135b0c3ab298f8d47e441c0416)